pulsar-broker/src/main/java/org/apache/pulsar/broker/service/schema/SchemaRegistryServiceImpl.java

```diff
@@ -242,17 +251,14 @@ private void checkCompatible(SchemaAndMetadata existingSchema, SchemaData newSch
         SchemaHash existingHash = SchemaHash.of(existingSchema.schema);
         SchemaHash newHash = SchemaHash.of(newSchema);
         SchemaData existingSchemaData = existingSchema.schema;
-        if (existingSchemaData.getType().isPrimitive()) {
-            if (newSchema.getType() != existingSchemaData.getType()) {
-                throw new IncompatibleSchemaException(String.format("Incompatible primitive schema: "
-                                + "exists schema type %s, new schema type %s",
-                        existingSchemaData.getType(), newSchema.getType()));
-            }
-        } else {
-            if (!newHash.equals(existingHash)) {
-                compatibilityChecks.getOrDefault(newSchema.getType(), SchemaCompatibilityCheck.DEFAULT)
-                        .checkCompatible(existingSchemaData, newSchema, strategy);
-            }
+        if (newSchema.getType() != existingSchemaData.getType()) {
```

Comment:
Don't you need to check if `strategy` is not `ALWAYS_COMPATIBLE`?

Reply:
There is no need to check compatibility here. Because in checkConsumerCompatibility will judge. If it is ALWAYS_COMPATIBLE, it will not be called here.